app/models/course/discussion/topic.rb

```diff
@@ -22,8 +22,31 @@ def self.global_topic_models
       where(actable_type: global_topic_models.map(&:name))
   end)
 
+  # Returns the topics from the user(s) specified.
+  # TODO: Remove the workaround and use the following implementation which is more cleaner after
+  # https://github.com/activerecord-hackery/squeel/issues/344 is closed, a similar rails issue is
+  # https://github.com/rails/rails/issues/20077.
+  #   scope :from_user, (lambda do |user_id|
+  #     where do
+  #       id >> Course::Assessment::Answer::ProgrammingFileAnnotation.from_user(user_id) |
+  #         id >> Course::Assessment::Answer.from_user(user_id)
+  #     end
+  #   end)
+  #
+  # @param[Integer|Array<Integer>] user_id, the id(s) of the user(s).
+  # @return[Array<Course::Discussion::Topic>]
+  scope :from_user, (lambda do |user_id|
+    where(
+      global_topic_models.map do |model|
```

Comment:
@allenwq why can't the scope be constructed within a block as @weiqingtoh suggested? I would have preferred that. The concrete models can be replaced with a `map`, like you did here.